src/Data/List/Relation/Binary/BagAndSetEquality.agda

```diff
@@ -115,7 +115,7 @@ module ⊆-Reasoning where
 module _ {a k} {A : Set a} {x y : A} {xs ys} where
 
   ∷-cong : x ≡ y → xs ∼[ k ] ys → x ∷ xs ∼[ k ] y ∷ ys
-  ∷-cong refl xs≈ys {y} =
+  ∷-cong P.refl xs≈ys {y} =
```

Comment:
Eugh the newly introduced clash with `refl` from `PropositionalEquality` is an unpleasant long-term problem and from a user perspective, we've managed to make the interface worse.

Perhaps we should consider prepending all the constructors with `↭-` to avoid this. A breaking change, but it does avoid this problem. The one wrinkle with that is that looking `↭-trans` in `Permutation.Propositional` it's actually doing something different from constructor `trans`. So we'd have to rename that as well...

Alternatively, as this isn't really going to bring the user any benefit, maybe we should simply prove an equivalence statement between `Setoid` instantiated with the propositional setoid and the `Propositional` definition and transfer the proofs that way?

Thoughts?

Reply:
> Thoughts?

I'm in general against adding prefixes (except for things like bundles where there are multiple
reasonable constructs with the name) because people can achieve the same result by importing
the module qualified.

Reply:
> I'm in general against adding prefixes (except for things like bundles where there are multiple
reasonable constructs with the name) because people can achieve the same result by importing
the module qualified.

Fair enough. It's a good point.

> I can swap the import strategy from qualifying PropositionalEquality's refl to
qualifiying Permutation's instead if you want.

I don't think that would change the fact that we're degrading the user experience to make our own life easier as library developers?

Reply:
See #2321 .
For whatever reason, Agda is (now!) happy for the `refl` of each of `PropositionalEquality` and `Permutation.Propositional` to coexist happily. Moreover the refactoring uncouples the dependency on inductive proofs on the latter relation.

Reply:
Resolved as far as this module is concerned.